docs/user-guide/continuous-aggregation/usecase-example.md

```diff
@@ -0,0 +1,122 @@
+
+2. **Real-time Monitoring**: A real-time monitoring system that continuously aggregates data from a stream of events and provides real-time alerts based on the aggregated data. For example, a system that aggregates data from a stream of sensor events and provides real-time alerts when the temperature exceeds a certain threshold.
+
+3. **Real-time Dashboard**: A real-time dashboard that shows the number of requests per minute, the average response time, and the number of errors per minute. This dashboard can be used to monitor the health of the system and to detect any anomalies in the system.
+
+In all these usecases, the continuous aggregation system continuously aggregates data from a stream of events and provides real-time insights and alerts based on the aggregated data. The system can also downsample the data to a lower resolution to reduce the amount of data stored and processed. This allows the system to provide real-time insights and alerts while keeping the data storage and processing costs low.
+
+## Real-time analytics example
+
+See [Overview](overview.md) for an example of real-time analytics. Which is to calculate the total number of logs, the minimum size, the maximum size, the average size, and the number of packets with the size greater than 550 for each status code in a 1-minute fixed window for access logs.
+
+Another example of real-time analytics is to get all distinct country from the `ngx_access_log` table. The query for continuous aggregation would be:
+
+```sql
+-- input table
+CREATE TABLE ngx_access_log (
+    client STRING,
+    country STRING,
+    access_time TIMESTAMP TIME INDEX
+);
+-- create flow task to calculate the distinct country
+CREATE FLOW calc_ngx_country
+SINK TO ngx_country
+AS
+SELECT
+    DISTINCT country,
+FROM ngx_access_log;
+```
+
+or if you want to group the data by time window, you can use the following query:
+
+```sql
+CREATE FLOW calc_ngx_country
+SINK TO ngx_country
+AS
+SELECT
+    DISTINCT country,
+    date_bin(INTERVAL '1 hour', access_time) as time_window,
+FROM ngx_access_log
+GROUP BY
+    country,
+    time_window;
+```
+
+The above query puts the data from the `ngx_access_log` table into the `ngx_country` table. It calculates the distinct country for each time window. The `date_bin` function is used to group the data into one-hour intervals. The `ngx_country` table will be continuously updated with the aggregated data, providing real-time insights into the distinct countries that are accessing the system. 
+
+Note that there is currently no persistent storage for flow's internal state(There is persistent storage for the table data however), so it's recommended to use appropriate time window to miniminize data loss, because if the internal state is lost, related time window data will be lost as well.
```

Comment:
cannot understand what is the flow's internal state, and what's the **appropriate** time window?